llvm/tools/llvm-exegesis/lib/Assembler.cpp

```diff
@@ -81,7 +81,7 @@ static bool generateSnippetSetupCode(const ExegesisTarget &ET,
       // If we're generating memory instructions, don't load in the value for
       // the register with the stack pointer as it will be used later to finish
       // the setup.
-      if (RV.Register == StackPointerRegister)
+      if (Register(RV.Register) == StackPointerRegister)
```

Comment:
I think the underlying type should probably be updated in `RegisterValue.h` rather than casting here?

Reply:
Yeah I wasn't super happy about this. StackPointer is probably always a physical register, but the interface it comes from uses Register. All the registers going into RegisterValue are all physical registers, so I wasn't sure about using Register there just to make these 2 compares compile.

In the long run maybe Register and MCRegister should be comparable by operator==.